src/Build/BackEnd/Components/SdkResolution/NuGet/GlobalJsonReader.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using SdkResolverContextBase = Microsoft.Build.Framework.SdkResolverContext;
+
+namespace Microsoft.Build.BackEnd.SdkResolution.NuGet
+{
+    /// <summary>
+    /// Reads MSBuild related sections from a global.json.
+    /// </summary>
+    internal static class GlobalJsonReader
+    {
+        public const string GlobalJsonFileName = "global.json";
+
+        public const string MSBuildSdksPropertyName = "msbuild-sdks";
+
+        /// <summary>
+        /// Walks up the directory tree to find the first global.json and reads the msbuild-sdks section.
+        /// </summary>
+        /// <returns>A <see cref="Dictionary{String,String}"/> of MSBuild SDK versions from a global.json if found, otherwise <code>null</code>.</returns>
+        public static Dictionary<string, string> GetMSBuildSdkVersions(SdkResolverContextBase context)
```

Comment:
What's the relationship between our finding and interpreting this file and `dotnet` CLI's? Are we doing anything to make sure they follow the same logic?

Reply:
Unfortunately, the code that finds and parses `global.json` in the dotnet CLI is in unmanaged code ([here](https://github.com/dotnet/core-setup/blob/77152705cd56e7d89f52585959ec3559d91084c4/src/corehost/cli/fxr/fx_muxer.cpp#L894)).

I followed the same overall pattern to find `global.json` but the logic could get out of sync.